Prepends header line to data logs

stulacy opened the pull request:
The CSV header from `misc/header.txt` is prepended to data files prior to being compressed.

A unit test is included, which contrary to general best practices creates temporary files and reads and writes to them.
This isn't ideal, but since the function itself needs to read 2 files and write to 1, it is an effective way of testing the behaviour without using hard to follow mocks. 

This commit also includes some styling changes introduced by `black` (using max line length of 80).

Closes #9

stulacy committed:
Prepends header line to data logs

This commit also includes some styling changes introduced by `black`
(max line length of 80).

stulacy committed:
fixed header path

kilicomu reviewed (changes requested):
I think you are adding the header in the wrong place! It probably wants to be in `DataWriter` - the uploading code is getting removed for release.

stulacy commented:
Since `DataWriter` is running as a background service, continuously writing serial data to a daily log file, it would be awkward to add both the header and compression in there. Wouldn't it make more sense to add the header and compress as part of a nightly process, so once the daily log file is no longer being written to, add header and compress? 
`upload.py` is basically doing this already - we can just cut out the S3 uploading functionality from it.

kilicomu commented:
Yep, just needs a rename to something appropriate.

On Fri, 20 Aug 2021 at 13:59, Stuart Lacy ***@***.***> wrote:

> Since DataWriter is running as a background service, continuously writing
> serial data to a daily log file, it would be awkward to add both the header
> and compression in there. Wouldn't it make more sense to add the header and
> compress as part of a nightly process, so once the daily log file is no
> longer being written to, add header and compress?
> upload.py is basically doing this already - we can just cut out the S3
> uploading functionality from it.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/wacl-york/bocs_control/pull/14#issuecomment-902675042>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACG5SE4IXIWRQCFCYP5L6ODT5ZGTXANCNFSM5COSGPYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Killian Murphy
Research Software Engineer

Wolfson Atmospheric Chemistry Laboratories
University of York
Heslington
York
YO10 5DD
+44 (0)1904 32 1223

e-mail disclaimer: http://www.york.ac.uk/docs/disclaimer/email.htm

stulacy committed:
Removed upload functionality

The release code will not upload any data to S3. Instead, the nightly
task will simply archive the previous day's data log file.

stulacy committed:
Use strftime

stulacy commented:
I've pushed changes that:

- Rename `upload.py` -> `archive.py`
- Removed uploading functionality from `archive.py`
- Updated the run script `run_upload.sh` to `run_archive.sh`. The site name is longer needed as an argument as it was only used in `upload.py` to grab the approprite AWS profile

stulacy committed:
More understandable file concatenation

kilicomu reviewed (commented):
Happy with this - needs testing on the BOCS before final approval.

stulacy committed:
Removed all references to uploading

stulacy commented:
> Happy with this - needs testing on the BOCS before final approval.

Have tested changes on Pi, header is being correctly added to the files when archiving. Have confirmed that the dimensions of the header matches the data rows both manually and also by using several CSV parsers (2 in R - `read_csv` and `read.csv` and `pandas.read_csv`). Have plotted the data using the headers embedded in the archived log files and the scale of the measurements match up with my expectations for the columns.
